controllers/gateway_controller_reconciler_utils.go

```diff
@@ -87,42 +87,58 @@ func (r *GatewayReconciler) ensureGatewayMarkedReady(ctx context.Context, gatewa
 
 		count := len(services)
 		if count > 1 {
+			gateway.Status.Addresses = []gatewayv1alpha2.GatewayAddress{}
 			return fmt.Errorf("found %d services for DataPlane currently unsupported: expected 1 or less", count)
```

Comment:
I do reckon you're not changing this line but while we're here, how about:

```suggestion
			return fmt.Errorf("found %d services for DataPlane %s/%s, currently unsupported: expected 1 or less",
				count, dataplane.Namespace, dataplane.Name,
			)
```